backend/src/auth/auth.service.ts

```diff
@@ -66,4 +71,17 @@ export class AuthService {
       );
     }
   }
+
+  /**
+   * Generates an username, repeated aslong as the generated username already exists
+   * @returns username a random string of up to 20 characters
+   * @private this method is only used in the auth service
+   */
+  private async generateUsername(): Promise<string> {
+    let username = generateUsername('', 0, 15);
+    while (await this.userService.checkUsernameExists(username)) {
+      username = generateUsername('', 0, 20);
```

Comment:
Why are you using different lengths?

Reply:
That was a copy/paste mistake from my side.